cli/command/stack/kubernetes/cli.go

```diff
@@ -69,9 +73,43 @@ func WrapCli(dockerCli command.Cli, opts Options) (*KubeCli, error) {
 	}
 	cli.clientSet = clientSet
 
+	if dockerCli.ClientInfo().HasAll() {
+		if err := cli.checkHostsMatch(); err != nil {
+			return nil, err
+		}
+	}
 	return cli, nil
 }
 
 func (c *KubeCli) composeClient() (*Factory, error) {
 	return NewFactory(c.kubeNamespace, c.kubeConfig, c.clientSet)
 }
+
+func (c *KubeCli) checkHostsMatch() error {
+	daemonEndpoint, err := url.Parse(c.Client().DaemonHost())
+	if err != nil {
+		return err
+	}
+	kubeEndpoint, err := url.Parse(c.kubeConfig.Host)
+	if err != nil {
+		return err
+	}
+	if daemonEndpoint.Hostname() == kubeEndpoint.Hostname() {
+		return nil
+	}
+	// The daemon can be local in Docker for Desktop, e.g. "npipe", "unix", ...
+	if daemonEndpoint.Scheme != "tcp" {
+		ips, err := net.LookupIP(kubeEndpoint.Hostname())
+		if err != nil {
+			return err
+		}
+		for _, ip := range ips {
+			if ip.IsLoopback() {
+				return nil
+			}
+		}
+	}
+	fmt.Fprintf(c.Out(), "Warning: Swarm and Kubernetes hosts do not match - DOCKER_HOST=%s and kubernetes context=%s -"+
+		" update DOCKER_HOST or use 'kubectl config use-context' to match\n", daemonEndpoint.Hostname(), kubeEndpoint.Hostname())
```

Comment:
let me think a bit about mentioning `DOCKER_HOST`, because this could also be because of `docker -H=<some host>`

Reply:
What about
```
"Warning: Swarm and Kubernetes hosts do not match - docker host=%s and kubernetes context=%s - update DOCKER_HOST (or pass -H) or use 'kubectl config use-context' to match```
?

Reply:
Yes that looks better. Perhaps put the "docker host / kubernetes host" in brackets, and align the output for readability (as it's printed when working with docker interactively, we should make it user-friendly) e.g.;


```
WARNING: Swarm and Kubernetes hosts do not match (docker host=%s, kubernetes host=%s)
         Update $DOCKER_HOST (or pass -H), or use 'kubectl config use-context' to match.
```

Should we use `context=` or `host=` for k8s? "Context" may be confused with the name of the context?

Reply:
`host` should be fine, I updated the message.